src/Illuminate/Support/Number.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace Illuminate\Support;
+
+use Illuminate\Support\Facades\App;
+use Illuminate\Support\Traits\Macroable;
+use NumberFormatter;
+use RuntimeException;
+
+class Number
+{
+    use Macroable;
+
+    /**
+     * Format the number according to the current locale.
+     *
+     * @param  float|int  $number
+     * @param  ?string  $locale
+     * @return false|string
+     */
+    public static function format($number, $locale = null)
+    {
+        static::needsIntlExtension();
+
+        $formatter = new NumberFormatter($locale ?? App::getLocale(), NumberFormatter::DECIMAL);
+
+        return $formatter->format($number);
+    }
+
+    /**
+     * Format the number to a fluent human-readable string.
+     *
+     * @param  float|int  $number
+     * @param  ?string  $locale
+     * @return false|string
+     */
+    public static function toHuman($number, $locale = null)
+    {
+        static::needsIntlExtension();
+
+        $formatter = new NumberFormatter($locale ?? App::getLocale(), NumberFormatter::SPELLOUT);
+
+        return $formatter->format($number);
+    }
+
+    /**
+     * Format the number to a percent format.
+     *
+     * @param  float|int  $number
+     * @param  int  $precision
+     * @param  string  $locale
+     * @return false|string
+     */
+    public static function toPercent($number, $precision = 2, $locale = 'en')
+    {
+        static::needsIntlExtension();
+
+        $formatter = new NumberFormatter($locale, NumberFormatter::PERCENT);
+
+        $formatter->setAttribute(NumberFormatter::MAX_FRACTION_DIGITS, $precision);
+
+        return $formatter->format($number / 100);
```

Comment:
I'm dividing the supplied value by 100 so that 50 = 50% as that's how Rails does it, and that's what Taylor linked to in his suggestion for this class. The default number formatter would consider 0.5 to be 50% and 50 to be 5000%. I'm not sure which option is best, so I went with the Rails format. https://api.rubyonrails.org/classes/ActionView/Helpers/NumberHelper.html#method-i-number_to_percentage